src/config.rs

```diff
@@ -150,6 +151,9 @@ pub struct Config {
 
     /// Default list of sources and the sources used for proxy mode.
     pub sources: Arc<Vec<SourceConfig>>,
+
+    /// Allow reserved IP addresses for requests to sources.
+    pub connect_to_reserved_ips: bool,
```

Comment:
Instead of making this just a bool, let's assume that everyone will want to avoid requests to localhost by default. Also, in Sentry we will need to override this with a bunch of network-internal IP ranges. That's why I would suggest to make this a list, and assign `RESERVED_IP_BLOCKS` (from `http.rs`) as `#[serde(default = "...")]`. Then, we can _always_ use the `SafeResolver` by default.

Reply:
`RESERVED_IP_BLOCKS` is a standard, see https://en.wikipedia.org/wiki/Reserved_IP_addresses. The value in sentry also corresponds to that standard, and ops themselves have talked about making it a bool in sentry because there is no reason to modify it.